### PR TITLE
feat: integration tests for pluggable auth

### PR DESCRIPTION
**Pull request by ScruffyProdigy**

### CODE REVIEW

File: .kokoro/presubmit/integration.cfg [hunk 1]
```diff
@@ -35,4 +35,9 @@ env_vars: {
 env_vars: {
   key: "GCS_BUCKET"
   value: "byoid-it-bucket"
+}
+
+env_vars: {
+  key: "GOOGLE_EXTERNAL_ACCOUNT_ALLOW_EXECUTABLES"
+  value: "1"
```
Comment (lsirac) [inline .kokoro/presubmit/integration.cfg:42]: add newline at end of file

File: .kokoro/nightly/integration.cfg [hunk 2]
```diff
@@ -40,3 +40,8 @@ env_vars: {
   key: "GCS_BUCKET"
   value: "byoid-it-bucket"
 }
+
+env_vars: {
+  key: "GOOGLE_EXTERNAL_ACCOUNT_ALLOW_EXECUTABLES"
+  value: "1"
+}
```
Comment (lsirac) [inline .kokoro/nightly/integration.cfg:47]: add newline at end of file

File: oauth2_http/javatests/com/google/auth/oauth2/ITWorkloadIdentityFederationTest.java [hunk 3]
```diff
@@ -86,7 +88,7 @@ void setup() throws IOException {
    * using the iamcredentials generateIdToken API. This will use the service account client ID as
    * the sub field of the token. This OIDC token will be used as the external subject token to be
    * exchanged for a GCP access token via GCP STS endpoint and then to impersonate the original
-   * service account key.
+   * service account key. Retrieves the OIDC token from an output file.
```
Comment (lsirac) [inline oauth2_http/javatests/com/google/auth/oauth2/ITWorkloadIdentityFederationTest.java:91]: output file?

File: oauth2_http/javatests/com/google/auth/oauth2/ITWorkloadIdentityFederationTest.java [hunk 4]
```diff
@@ -178,6 +197,57 @@ private GenericJson buildIdentityPoolCredentialConfig() throws IOException {
...
+
+    GenericJson executableJson = new GenericJson();
+    executableJson.setFactory(OAuth2Utils.JSON_FACTORY);
+    executableJson.put("success", true);
+    executableJson.put("version", 1);
+    executableJson.put("expiration_time", expiration_time.toEpochMilli());
+    executableJson.put("token_type", "urn:ietf:params:oauth:token-type:jwt");
+    executableJson.put("id_token", idToken);
+
+    StringWriter fileContents = new StringWriter();
```
Comment (lsirac) [inline oauth2_http/javatests/com/google/auth/oauth2/ITWorkloadIdentityFederationTest.java:213]: why use StringWriter?

**Review by lsirac — APPROVED** [review]

**Review by TimurSadykov — APPROVED** [review]
LGTM, please address Leo comments

File: .kokoro/presubmit/integration.cfg (hunk 1 above)
Comment (ScruffyProdigy) [reply .kokoro/presubmit/integration.cfg:42]: Done

File: .kokoro/nightly/integration.cfg (hunk 2 above)
Comment (ScruffyProdigy) [reply .kokoro/nightly/integration.cfg]: Done

File: oauth2_http/javatests/com/google/auth/oauth2/ITWorkloadIdentityFederationTest.java (hunk 3 above)
Comment (ScruffyProdigy) [reply oauth2_http/javatests/com/google/auth/oauth2/ITWorkloadIdentityFederationTest.java]: Now just "file"

File: oauth2_http/javatests/com/google/auth/oauth2/ITWorkloadIdentityFederationTest.java (hunk 4 above)
Comment (ScruffyProdigy) [reply oauth2_http/javatests/com/google/auth/oauth2/ITWorkloadIdentityFederationTest.java]: StringBuilder now

**Pull request merged**